File: kt1/KT1.py
class Liige:
    def __init__(self, nimi, vanus, kestus):
        self.nimi = nimi
        self.vanus = vanus
        self.kestus = kestus
    
    def get_teenused(self):
        raise NotImplementedError("Meetod peab olema ülekirjutatud alamaklassis")
    
    def __str__(self):
        return f"{self.nimi}, {self.vanus}, {self.kestus} kuud, {self.get_teenused()}"
    
    def __lt__(self, other):
        return self.kestus < other.kestus

class Tavaliige(Liige):
    def __init__(self, nimi, vanus, kestus, teenus):
        super().__init__(nimi, vanus, kestus)
        self.teenus = teenus
    
    def get_teenused(self):
        return self.teenus
    
    def __str__(self):
        return f"Tavaliige: {super().__str__()}"

class Preemiumliige(Liige):
    def __init__(self, nimi, vanus, kestus, teenused):
        super().__init__(nimi, vanus, kestus)
        self.teenused = teenused
    
    def get_teenused(self):
        return ", ".join(self.teenused)
    
    def __str__(self):
        return f"Preemiumliige: {super().__str__()}"

class Spordiklubi:
    def __init__(self, nimi):
        self.nimi = nimi
        self.liikmed = []
    
    def lisa_liige(self, liige):
        self.liikmed.append(liige)
    
    def pikaajalised_liikmed(self):
        return sorted([liige for liige in self.liikmed if liige.kestus >= 12], key=lambda x: x.kestus, reverse=True)
    
    def __str__(self):
        return f"Spordiklubi {self.nimi} - {len(self.liikmed)} liiget"

def loe_liikmed_failist(failinimi):
    liikmed = []
    try:
        with open(failinimi, "r", encoding="utf-8") as f:
            for rida in f:
                andmed = rida.strip().split(";")
                liikme_tüüp, nimi, vanus, kestus, teenused = andmed[0], andmed[1], int(andmed[2]), int(andmed[3]), andmed[4]
                if liikme_tüüp == "Tavaliige":
                    liikmed.append(Tavaliige(nimi, vanus, kestus, teenused))
                elif liikme_tüüp == "Preemiumliige":
                    liikmed.append(Preemiumliige(nimi, vanus, kestus, teenused.split(",")))
    except Exception as e:
        print(f"Viga faili lugemisel: {e}")
    return liikmed

if __name__ == "__main__":
    spordiklubi = Spordiklubi("MyFitness")
    liikmed = loe_liikmed_failist("liikmed.txt")
    for liige in liikmed:
        spordiklubi.lisa_liige(liige)
    
    print(spordiklubi)
    print("\nKõik liikmed:")
    for liige in spordiklubi.liikmed:
        print(liige)
    
    print("\nPikaajalised liikmed:")
    for liige in spordiklubi.pikaajalised_liikmed():
        print(liige)

File: kt1/KT1.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

abstract class Liige implements Comparable<Liige> {
    protected String nimi;
    protected int vanus;
    protected int kestus;
    
    public Liige(String nimi, int vanus, int kestus) {
        this.nimi = nimi;
        this.vanus = vanus;
        this.kestus = kestus;
    }

    public abstract String teenused();
    
    @Override
    public int compareTo(Liige other) {
        return Integer.compare(this.kestus, other.kestus);
    }
}

class Tavaliige extends Liige {
    private String teenus;
    
    public Tavaliige(String nimi, int vanus, int kestus, String teenus) {
        super(nimi, vanus, kestus);
        this.teenus = teenus;
    }
    
    @Override
    public String teenused() {
        return "Tavaliige teenus: " + teenus;
    }
}

class Preemiumliige extends Liige {
    private List<String> teenused;
    
    public Preemiumliige(String nimi, int vanus, int kestus, List<String> teenused) {
        super(nimi, vanus, kestus);
        this.teenused = teenused;
    }
    
    @Override
    public String teenused() {
        return "Preemiumliige teenusteks on: " + String.join(", ", teenused);
    }
}

class Spordiklubi {
    private String nimi;
    private List<Liige> liikmed;
    
    public Spordiklubi(String nimi) {
        this.nimi = nimi;
        this.liikmed = new ArrayList<>();
    }
    
    public void lisaLiige(Liige liige) {
        liikmed.add(liige);
    }
    
    public List<Liige> pikaajalisedLiikmed() {
        List<Liige> pikaajalised = new ArrayList<>();
        for (Liige liige : liikmed) {
            if (liige.kestus >= 12) {
                pikaajalised.add(liige);
            }
        }
        return pikaajalised;
    }
    
    public void kuvaLiikmed() {
        for (Liige liige : liikmed) {
            System.out.println(liige.nimi + " - " + liige.teenused());
        }
    }
}

public class Peaklass {
    public static List<Liige> loeLiikmed(String failiNimi) throws IOException {
        List<Liige> liikmed = new ArrayList<>();
        List<String> read = Files.readAllLines(Paths.get(failiNimi));
        for (String rida : read) {
            String[] osad = rida.split(";");
            String liikmeTüüp = osad[0];
            String nimi = osad[1];
            int vanus = Integer.parseInt(osad[2]);
            int kestus = Integer.parseInt(osad[3]);
            if (liikmeTüüp.equals("Tavaliige")) {
                liikmed.add(new Tavaliige(nimi, vanus, kestus, osad[4]));
            } else if (liikmeTüüp.equals("Preemiumliige")) {
                List<String> teenused = Arrays.asList(osad[4].split(","));
                liikmed.add(new Preemiumliige(nimi, vanus, kestus, teenused));
            }
        }
        return liikmed;
    }
    
    public static void main(String[] args) {
        try {
            List<Liige> liikmed = loeLiikmed("liikmed.txt");
            Spordiklubi klubi = new Spordiklubi("Spordiklubi X");
            for (Liige liige : liikmed) {
                klubi.lisaLiige(liige);
            }
            System.out.println("Kõik liikmed:");
            klubi.kuvaLiikmed();
            System.out.println("\nPikaajalised liikmed:");
            for (Liige liige : klubi.pikaajalisedLiikmed()) {
                System.out.println(liige.nimi);
            }
        } catch (IOException e) {
            System.err.println("Viga faili lugemisel: " + e.getMessage());
        }
    }
}